Automattic/wp-calypso
# Imports: Move actions out of SiteImporterInputPane

#### Changes proposed in this Pull Request

This PR looks to pull out (almost) all action/state logic out in to dedicated actions/reducer files, handling some state directly through redux but allowing the flux store to work as before.
One of the main motivations for this was that prior to these changes, we had `componentWillReceiveProps` watching for prop changes and dispatching actions in response to those changes. The trouble with this, is that we lose control of state flow this way. If we decide to create a new composition of actions we would still see other unwanted actions triggered by this component.

#### Testing instructions

We need to test the whole site importer flow:

- Start at http://calypso.localhost:3000/settings/import/
- Choose the Wix importer and enter a valid wix site url (example: `https://smt593.wixsite.com/wowz/static-page`)
- Hit enter and go through the flow as usual - confirm the site preview step and watch the import take place.
- It should, of course, get through to the import completed state without issue, you may notice the odd flicker though. I think this is just a bigger problem caused by us using a faked local ID until we have a proper one from the server. Because the ID essentially changes midway through the import process the first importer component will unmount and be immediately replaced with another, keyed by the new ID.
- Try the flow again, but hit the cancel button at different stages - they should work as before.
